doc/jinja-filters.md

```diff
@@ -0,0 +1,438 @@
+("2017-12-31", "2018-08-16", "MMMM YYYY")           "<span class='date'>December 2017</span> to <span class='date'>August 2018</span>"
+("2018-08-14", "2018-08-16", "MMMM YYYY")           "<span class='date'>August 2018</span> to <span class='date'>August 2018</span>"
+("2017-12-31", "2018-08-16", "YYYY")                "<span class='date'>2017</span> to <span class='date'>2018</span>"
+("2017-07-31", "2018-08-16", "YYYY")                "<span class='date'>2017</span> to <span class='date'>2018</span>"
+("2018-08-14", "2018-08-16", "EEEE d")              "<span class='date'>Tuesday 14</span> to <span class='date'>Thursday 16</span>"
+```
+
+
+## first_non_empty_item
+Returns the first non empty value in a list of values.
+
+##### Parameters: 
+- *items: List of items where some of them could have no value.
+
+##### Context:
+Implemented to replace trad_as_or_runame hard coding. 
+```
+Did any significant changes occur to the total retail turnover for {{ first_non_empty_item(metadata['trad_as'], metadata['ru_name']) }}?
+```
+
+##### Examples:
+```
+Input:                          Output:
+("", "Second", "", "Fourth")    "Second"
+("", Undefined(), None, "0")    "0"
+('', False, None, Undefined())  "False"
+```
+
+
+## format_household_name
+Concatenates a list of names into a single space separated string.
+
+##### Parameters: 
+- names: List of names to concatenate
+
+##### Context:
+To display a person's full name when the answers are entered as first, middle and last name. 
+```
+Are you <em>{{ [answers['first-name'][group_instance], answers['middle-names'][group_instance], answers['last-name'][group_instance]] | format_household_name }}<em>?
+```
+
+##### Examples:
+```
+Input:              Output:
+["John", "Doe"]     "John Doe"
+["John", ""]        "John"
+["", "Doe"]         "Doe"
+```
+
+
+## format_household_name_possessive
+Concatenates a list of names into a single space separated string and adds the possessive to the end (’s or just ’).
+
+##### Parameters: 
+- names: List of names to concatenate
+
+##### Context:
+To display a person's full name when the answers are entered as first, middle and last name but also to express a person's possession within the question. 
+```
+What is {{[answers['first-name'][group_instance], answers['last-name'][group_instance]] | format_household_name_possessive }} date of birth?
+```
+
+##### Examples:
+```
+Input:              Output:
+["John", "Doe"]     "John Doe's"
+["John", "Jones"]   "John Jones'"
+```
+
+## format_unordered_list
+Wraps a list of items in "ul" and "li" tags.
+
+##### Parameters: 
+- list_items: List of items to be wrapped into a html list
+
+##### Context:
+Lists selected pay patterns for MWSS. Used as a description: 
+```
+{{ answers['pay-pattern-frequency-answer']|format_unordered_list }}
+```
+
+##### Examples:
+```
+Input:                  Output:
+[['item 1', 'item 2']]  "<ul><li>item 1</li><li>item 2</li></ul>"
```

Comment:
Is the list within a list correct?

Reply:
Added card https://trello.com/c/DrZyF3sP to address this.